src/components/grid-selector.js

```diff
@@ -0,0 +1,57 @@
+import React, { useEffect, useRef } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+function GridSelector({ items, selectedItemKey, onSelectionChange }) {
+  const listRef = useRef();
+
+  useEffect(() => {
+    const selectedElement = listRef.current?.querySelector(`[data-item-key="${selectedItemKey.replaceAll('"', '\\"')}"]`);
+    if (selectedElement && selectedElement !== document.activeElement) {
+      selectedElement.focus();
+    }
+  }, [selectedItemKey]);
+
+  const handleFocusCapture = event => {
+    const itemKey = event.target.getAttribute('data-item-key');
+    if (itemKey && itemKey !== selectedItemKey) {
+      onSelectionChange?.(itemKey, false);
```

Comment:
Elvis can leave the building, since `onSelectionChange` defaults to empty function.

Reply:
Elvis has stopped rocking the house, thanks!

Reply:
🕺